lib/bolt/transpiler.rb

```diff
@@ -0,0 +1,68 @@
+# frozen_string_literal: true
+
+require 'bolt/error'
+require 'bolt/pal/yaml_plan'
+require 'bolt/pal/yaml_plan/loader'
+require 'bolt/util'
+
+module Bolt
+  class YamlTranspiler
+    class ConvertError < Bolt::Error
+      def initialize(msg, plan_path)
+        super(msg, 'bolt/convert-error', { "plan_path" => plan_path })
+      end
+    end
+
+    def self.transpile(plan_path)
+      @plan_path = plan_path
+      @module = Bolt::Util.module_name(plan_path)
```

Comment:
This should probably expand the plan_path first. It currently fails if you try to run it from inside a module.